rsocket/RSocketClient.cpp

```diff
@@ -123,10 +123,11 @@ folly::Future<folly::Unit> RSocketClient::resume() {
 
 void RSocketClient::disconnect(folly::exception_wrapper ex) {
   CHECK(stateMachine_);
-  evb_->runInEventBaseThread([ this, ex = std::move(ex) ]() mutable {
-    VLOG(2) << "Disconnecting RSocketStateMachine on EventBase";
-    stateMachine_->disconnect(std::move(ex));
-  });
+  evb_->runInEventBaseThread(
```

Comment:
lgtm. Any reason why we are not similarly capturing (token_, protocolVersion_) for resume()

Reply:
We call createState() in resume, which is a method on this, so we'll need some more thought about how we want to handle the resume case. Possibly just say "do not release the unique pointer to the RSocketClient until after the resume has completed".